docs/garage.md

```diff
@@ -0,0 +1,246 @@
+# Using Garage Object Storage in Bento
+
+This document covers how to use [Garage](https://garagehq.deuxfleurs.fr/), a lightweight, self-hosted, S3-compatible object storage solution, which replaces MinIO in Bento deployments which use a local S3 API provider.
+
+## Architecture
+
+Garage exposes three ports for different purposes:
+
+| Port     | Purpose        | Used By                              | Protocol                  |
+| -------- | -------------- | ------------------------------------ | ------------------------- |
+| **3900** | S3 API         | Services (DRS, Drop-Box)             | HTTP REST (S3-compatible) |
+| **3901** | RPC (Internal) | Garage nodes (cluster communication) | Custom RPC                |
+| **3903** | Admin API      | Administration and management        | HTTP REST                 |
+
+In Bento's single-node configuration (`replication_factor = "1"`), the RPC port is still required for internal operations.
+
+## Quick Start
+
+### 1. Initialize Docker Networks, Directories and Certs
+
+```bash
+./bentoctl.bash init-docker
+./bentoctl.bash init-dirs
+./bentoctl.bash init-certs  # use -f flag if certs directory already exists
+./bentoctl.bash restart gateway
+```
+
+This creates the required `bentov2-garage-net` network.
+
+> **Note**: If you see "all predefined address pools have been fully subnetted", run `docker network prune -f` to clean up unused networks, then retry.
+
+### 2. Add Garage Domain to /etc/hosts
+
+Add the following entry to your `/etc/hosts` file for local access:
+
+```bash
+127.0.0.1  garage.bentov2.local
+127.0.0.1  admin.garage.bentov2.local
+```
+
+### 3. Configure Garage Variables
+
+Edit your `local.env` file and set the following variables:
+
+```bash
+# local.env
+
+# Enable Garage profile
+BENTO_GARAGE_ENABLED='true'
+
+# Generate RPC secret (must be exactly 64 hexadecimal characters)
+# Use: openssl rand -hex 32
+BENTO_GARAGE_RPC_SECRET='<your-64-char-hex-secret>'
+
+# Set admin token (use a secure value for production)
+BENTO_GARAGE_ADMIN_TOKEN='<your-secure-admin-token>'
+```
+
+**Generate the RPC secret:**
```

Comment:
this is kind of out of order with the other sub-steps here - I need it before this line tells me how (except for the comment already there)